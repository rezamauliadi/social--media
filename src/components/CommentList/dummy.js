export default [
  {
    postId: 1,
    id: 1,
    name: "id labore",
    email: "Eliseo@gardner.biz",
    body: "laudantium "
  },
  {
    postId: 1,
    id: 2,
    name: "quo vero ",
    email: "Jayne_Kuhic@sydney.com",
    body: "est natus enim nihil est "
  },
  {
    postId: 1,
    id: 3,
    name: "odio adipisci",
    email: "Nikita@garfield.biz",
    body: "quia molestiae reprehenderit "
  }
];
