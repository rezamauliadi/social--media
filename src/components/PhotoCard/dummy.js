const albums = [
  {
    userId: 1,
    id: 1,
    title: "quidem molestiae enim"
  },
  {
    userId: 2,
    id: 11,
    title: "quidem molestiae enim"
  }
];
const photos = [
  {
    albumId: 1,
    id: 1,
    title: "accusamus beatae ad facilis cum similique qui sunt",
    url: "https://via.placeholder.com/600/92c952",
    thumbnailUrl: "https://via.placeholder.com/150/92c952"
  },
  {
    albumId: 2,
    id: 51,
    title: "non sunt voluptatem placeat consequuntur rem incidunt",
    url: "https://via.placeholder.com/600/8e973b",
    thumbnailUrl: "https://via.placeholder.com/150/8e973b"
  }
];

export default { albums, photos };
