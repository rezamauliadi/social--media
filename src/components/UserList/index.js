import React from "react";
import PropTypes from "prop-types";

import { Card, Image } from "semantic-ui-react";

import getAvatar from "src/helpers/avatar-helper";

const UserList = ({ users }) => {
  const userCard = user => (
    <Card key={user.id} style={{ width: "198px" }}>
      <Image src={getAvatar(user.id)} ui={false} wrapped />
      <Card.Content>
        <Card.Header>{user.name}</Card.Header>
        <Card.Meta>
          <span>{user.username}</span>
        </Card.Meta>
        <Card.Description>{user.company.catchPhrase}</Card.Description>
      </Card.Content>
    </Card>
  );

  return (
    <div>
      <Card.Group>{users.map(user => userCard(user))}</Card.Group>
    </div>
  );
};

UserList.propTypes = {
  users: PropTypes.array
};

export default UserList;
