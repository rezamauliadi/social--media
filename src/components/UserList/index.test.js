import React from "react";
import { shallow } from "enzyme";

import UserList from "./index.js";
import dummy from "./dummy.js";

describe("Component: UserList", () => {
  const users = dummy;
  const url = "/users";

  it("should match snapshot", () => {
    const props = { users, url };
    const component = shallow(<UserList {...props} />);

    expect(component).toMatchSnapshot();
  });
});
