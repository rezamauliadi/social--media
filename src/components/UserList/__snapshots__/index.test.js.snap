// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: UserList should match snapshot 1`] = `
<div>
  <CardGroup>
    <Card
      as={[Function]}
      key="1"
      style={
        Object {
          "width": "198px",
        }
      }
      to="/users/1/posts"
    >
      <Image
        as="img"
        src="/images/daniel.jpg"
        ui={false}
        wrapped={true}
      />
      <CardContent>
        <CardHeader>
          Leanne Graham
        </CardHeader>
        <CardMeta>
          <span>
            Bret
          </span>
        </CardMeta>
        <CardDescription>
          Multi-layered client-server neural-net
        </CardDescription>
      </CardContent>
    </Card>
    <Card
      as={[Function]}
      key="2"
      style={
        Object {
          "width": "198px",
        }
      }
      to="/users/2/posts"
    >
      <Image
        as="img"
        src="/images/jenny.jpg"
        ui={false}
        wrapped={true}
      />
      <CardContent>
        <CardHeader>
          Ervin Howell
        </CardHeader>
        <CardMeta>
          <span>
            Antonette
          </span>
        </CardMeta>
        <CardDescription>
          Proactive didactic contingency
        </CardDescription>
      </CardContent>
    </Card>
  </CardGroup>
</div>
`;
